src/BuildingRegistry/Building/BuildingUnit_State.cs

```diff
@@ -271,5 +272,13 @@ private void When(BuildingUnitAddressWasDetachedBecauseAddressWasRemoved @event)
 
             _lastEvent = @event;
         }
+
+        private void When(BuildingUnitAddressWasReplacedBecauseAddressWasReaddressed @event)
+        {
+            _addressPersistentLocalIds.Remove(new AddressPersistentLocalId(@event.SourceAddressPersistentLocalId));
```

Comment:
I would check if the address to remove still exists in the list and if the address to add is not already in the list.